display/more.rs

```diff
@@ -0,0 +1,1963 @@
+            .chunks(line_len)
+            .map(|ss| String::from_iter(ss))
+            .collect::<Vec<String>>(),
+        ":".repeat(header_width))
+    }else{
+        (vec![name_and_ext.clone()], ":".repeat(name_and_ext.len() + 4))
+    };
+
+    name_and_ext.insert(0, border.clone());
+    name_and_ext.push(border);
+    name_and_ext
+}
+
+/// Get char from [`&str`] by its position
+fn get_char(string: &str, position: usize) -> Option<char>{
+    string.chars().nth(position)
+}
+
+/// Parses [`String`] into [`Command`] and returns result with reminder
+fn parse(commands_str: String) -> Result<(Command, String, Command), MoreError>{
+    let mut command = Command::UnknownCommand;
+    let mut count: Option<usize> = None;
+    let mut next_possible_command = Command::UnknownCommand;
+    
+    let mut i = 0;
+    let mut chars = commands_str.chars();
+    let commands_str_len= commands_str.len();
+
+    while command == Command::UnknownCommand && i < commands_str_len{
+        let Some(ch) = chars.next() else { break; };
+        command = match ch{
+            ch if ch.is_numeric() => {
+                let mut count_str = String::new();
+                loop{
+                    let Some(ch) = get_char(&commands_str, i) else { break; };
+                    if !ch.is_numeric(){ break; }
+                    count_str.push(ch);
+                    i += 1;
+                }
+                if let Ok(new_count) = count_str.parse::<usize>(){
+                    let _ = chars.by_ref().skip(count_str.len());
+                    count = Some(new_count);
+                }
+                continue;
+            },
+            'h' => Command::Help,
+            'f' | '\x06' => Command::ScrollForwardOneScreenful(count),
+            'b' | '\x02' => Command::ScrollBackwardOneScreenful(count),
+            ' ' => Command::ScrollForwardOneLine{ count, is_space: true},
+            'j' | '\n' => Command::ScrollForwardOneLine{ count, is_space: false },
+            'k' => Command::ScrollBackwardOneLine(count),
+            'd' | '\x04' => Command::ScrollForwardOneHalfScreenful(count),
+            's' => Command::SkipForwardOneLine(count),
+            'u' | '\x15' => Command::ScrollBackwardOneHalfScreenful(count),
+            'g' => Command::GoToBeginningOfFile(count),
+            'G' => Command::GoToEOF(count),
+            'r' | '\x0C' => Command::RefreshScreen,
+            'R' => Command::DiscardAndRefresh,
+            'm' => {
+                i += 1;
+                let Some(ch) = get_char(&commands_str, i) else { break; };
+                if ch.is_ascii_lowercase() {
+                    Command::MarkPosition(ch)
+                }else{
+                    next_possible_command = Command::MarkPosition(' ');
+                    Command::UnknownCommand
+                }
+            },
+            '/' => {
+                i += 1;
+                let Some(ch) = get_char(&commands_str, i) else { break; };
```

Comment:
You can just [`peek`](https://doc.rust-lang.org/std/iter/struct.Peekable.html#method.peek): `chars.peek()`
Same for other `i += 1; next_char`